app/models/miq_request.rb

```diff
@@ -17,7 +17,7 @@ class MiqRequest < ApplicationRecord
   belongs_to :parent, :class_name => "MiqRequest"
   has_many   :miq_approvals,     :dependent   => :destroy
   has_many   :miq_request_tasks, :dependent   => :destroy
-  has_many   :request_logs,      :dependent   => :destroy
+  has_many   :request_logs, as: :resource
```

Comment:
Please use the hash-rocket style for consistency.

```suggestion
  has_many   :request_logs, :as => :resource
```

Reply:
Also I think you want to keep the dependent destroy here, right?

Reply:
I understand the dependent destroy being removed here, but I how do the records gets removed?  Should there be a separate "all_request_logs" or something?

Reply:
Yes, I think I have to return the request_id to the schema. It will be helpful for the view and the model.

Reply:
Yeah, I didn't realize that was to be used for a different purpose.  Sounds good.